### PR TITLE
Cleanup Globalization info code

### PR DESCRIPTION
**Pull request by hughbe**
And fix some param names

### CODE REVIEW

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 1]
```diff
@@ -974,8 +954,10 @@ internal string SENGDISPLAYNAME
             }
         }
 
-        // Native pretty name for this locale (ie: Deutsch (Deutschland))
-        internal string SNATIVEDISPLAYNAME
```
Comment (filipnavara) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: Need `CultureData.Windows.cs` updated to use the new names too.

File: src/System.Private.CoreLib/shared/System/Globalization/CharUnicodeInfo.cs [hunk 2]
```diff
@@ -302,26 +253,18 @@ internal static byte InternalGetCategoryValue(int ch, int offset)
...
         }
 
-        ////////////////////////////////////////////////////////////////////////
-        //
-        //Action: Returns the Unicode Category property for the character c.
-        //Returns:
-        //  an value in UnicodeCategory enum
-        //Arguments:
-        //  value  a Unicode String
-        //  index  Index for the specified string.
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CharUnicodeInfo.cs]: why didn't keep the params here?

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 3]
```diff
@@ -812,27 +802,15 @@ internal string CultureName
...
-                return _bUseOverrides;
-            }
-        }
+        /// <summary>
+        /// Are overrides enabled?
+        /// </summary>
+        internal bool UseUserOverride => _bUseOverrides;
 
-        // locale name (ie: de-DE, NO sort information)
-        internal string SNAME
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: >SNAME [](start = 24, length = 5)

I am ok to change the names here but these was named according to Windows LCID type names which was easily from the name of the properties to know what we are calling from Windows. 

As I mentioned I am ok because we support non Windows platforms now.

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 4]
```diff
@@ -812,27 +802,15 @@ internal string CultureName
...
-                return _sName;
-            }
-        }
+        /// <summary>
+        /// locale name (ie: de-DE, NO sort information)
+        /// </summary>
+        internal string Name => _sName ?? string.Empty;
 
         // Parent name (which may be a custom locale/culture)
         internal string SPARENT
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: >SPARENT [](start = 24, length = 7)

why did you change some and left the rest? I am not sure we need to have a mix from the modern and old names here.

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs [hunk 5]
```diff
@@ -188,110 +164,94 @@ private CultureInfo Culture
             }
         }
 
-        // TODO: This ignores other cultures that might want to do something similar
         private string LanguageName
         {
             get
             {
+                // TODO: This ignores other cultures that might want to do
+                // something similar
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: we may delete this comment

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs [hunk 6]
```diff
@@ -188,110 +164,94 @@ private CultureInfo Culture
...
 
-        ////////////////////////////////////////////////////////////////////////////
-        //
-        // Create an array of string which contains the abbreviated day names.
-        //
-        ////////////////////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Create an array of string which contains the abbreviated day names.
+        /// </summary>
+        private string[] InternalGetAbbreviatedDayOfWeekNames() => this.abbreviatedDayNames ?? InternalGetAbbreviatedDayOfWeekNamesCore();
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: >this. [](start = 67, length = 5)

do we need 'this' here?

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs [hunk 7]
```diff
@@ -188,110 +164,94 @@ private CultureInfo Culture
...
-        // Arguments:
-        //  None
-        // Exceptions:
-        //  None
-        //
-        ////////////////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Returns the string array of the one-letter day of week names.
+        /// </summary>
+        private string[] InternalGetSuperShortDayNames() => this.m_superShortDayNames ?? InternalGetSuperShortDayNamesCore();
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: >this. [](start = 60, length = 5)

same 'this' comment

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs [hunk 8]
```diff
@@ -307,21 +267,39 @@ internal DateTimeFormatInfo(CultureData cultureData, Calendar cal)
...
         }
 
         private void InitializeOverridableProperties(CultureData cultureData, CalendarId calendarId)
         {
             Debug.Assert(cultureData != null);
             Debug.Assert(calendarId != CalendarId.UNINITIALIZED_VALUE, "[DateTimeFormatInfo.Populate] Expected initalized calendarId");
 
-            if (this.firstDayOfWeek == -1) { this.firstDayOfWeek = cultureData.IFIRSTDAYOFWEEK; }
-            if (this.calendarWeekRule == -1) { this.calendarWeekRule = cultureData.IFIRSTWEEKOFYEAR; }
+            if (this.firstDayOfWeek == -1)
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: do we need 'this' in the whole method?

**Comment by tarekgh** [comment]
>                 if (this.optionalCalendars == null)

'this'

---
Refers to: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs:509 in 409ae2f. [](commit_id = 409ae2f5fe2f4c706810716840e6a4a4f9dba465, deletion_comment = False)

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs [hunk 9]
```diff
@@ -1707,50 +1645,40 @@ private static string[] GetMergedPatterns(string[] patterns, string defaultPatte
...
+        /// Default string isn't necessarily in our string array, so get the
+        /// merged patterns of both
+        /// </summary>
         private string[] AllYearMonthPatterns
         {
-            get
-            {
-                return GetMergedPatterns(this.UnclonedYearMonthPatterns, this.YearMonthPattern);
-            }
+            get => GetMergedPatterns(this.UnclonedYearMonthPatterns, this.YearMonthPattern);
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: >get [](start = 12, length = 3)

do we need get at all here? this apply to similar changes like that

File: src/System.Private.CoreLib/shared/System/Globalization/NumberFormatInfo.cs [hunk 10]
```diff
@@ -2,79 +2,45 @@
...
 namespace System.Globalization
 {
-    //
-    // Property             Default Description
-    // PositiveSign           '+'   Character used to indicate positive values.
-    // NegativeSign           '-'   Character used to indicate negative values.
-    // NumberDecimalSeparator '.'   The character used as the decimal separator.
-    // NumberGroupSeparator   ','   The character used to separate groups of
-    //                              digits to the left of the decimal point.
-    // NumberDecimalDigits    2     The default number of decimal places.
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/NumberFormatInfo.cs]: why removed this comment?

File: src/System.Private.CoreLib/shared/System/Globalization/StringInfo.cs [hunk 11]
```diff
@@ -2,103 +2,69 @@
...
 
-        public override bool Equals(object value)
+        public StringInfo(string value)
         {
-            if (value is StringInfo that)
-            {
-                return (_str.Equals(that._str));
-            }
-            return (false);
+            String = value;
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/StringInfo.cs]: >String [](start = 12, length = 6)

I think keeping 'this' here is a little bit better to distinguish it from String type

File: src/System.Private.CoreLib/shared/System/Globalization/TextInfo.cs [hunk 12]
```diff
@@ -150,43 +124,28 @@ internal void SetReadOnlyState(bool readOnly)
...
-            get
-            {
-                if (_listSeparator == null)
-                {
-                    _listSeparator = _cultureData.SLIST;
-                }
-                return _listSeparator;
-            }
-
+            get => _listSeparator ?? _cultureData.ListSeparator;
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/TextInfo.cs]: >_cultureData.ListSeparator; [](start = 37, length = 27)

you need to initialize _listSeparator if it is null

**Comment by tarekgh** [comment]
Thanks @hughbe for that great work. 

there is some expected failures here too because of the change of the exception parameter names. 

```
04:22:28       Assert.Equal() Failure
04:22:28                 ↓ (pos 0)
04:22:28       Expected: String
04:22:28       Actual:   value
04:22:28                 ↑ (pos 0)
04:22:28       Stack Trace:
04:22:28 Starting:    System.Linq.Queryable.Tests
04:22:28             at System.AssertExtensions.Throws[T](String paramName, Func`1 testCode)
04:22:28          /root/corefx-1949935/src/System.Globalization/tests/StringInfo/StringInfoString.cs(26,0): at System.Globalization.Tests.StringInfoString.String_Set_Invalid()
04:22:28    System.Globalization.Tests.NumberFormatInfoCurrencyNegativePattern.CurrencyNegativePattern_Set_Invalid [FAIL]
04:22:28       Assert.Equal() Failure
04:22:28                 ↓ (pos 0)
04:22:28       Expected: CurrencyNegativePattern
04:22:28       Actual:   value
04:22:28                 ↑ (pos 0)
04:22:28       Stack Trace:
04:22:28             at System.AssertExtensions.Throws[T](String paramName, Func`1 testCode)
04:22:28          /root/corefx-1949935/src/System.Globalization/tests/NumberFormatInfo/NumberFormatInfoCurrencyNegativePattern.cs(65,0): at
```

**Comment by tarekgh** [comment]
@hughbe could you please try to run the corefx tests against your coreclr private bits and then submit a PR in corefx repo fixing these tests? we need to do that from now to avoid breaking corefx when trying to bring the change to corefx. 

Note that, when fixing the test you need to ensure netcoreapp and NetFx runs successfully as we cannot change the exception parameters on NetFX (the full framework).

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs (hunk 3 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: Yeah it makes sense to name them appropriately that makes sense - I've updated all the remaining occurences

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs (hunk 4 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: Fixed all

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: Fixed

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs (hunk 6 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: Fixed here and elsewhere

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs (hunk 7 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: Ditto

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs (hunk 8 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: Fixed

File: src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs (hunk 9 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/DateTimeFormatInfo.cs]: Fixed

File: src/System.Private.CoreLib/shared/System/Globalization/NumberFormatInfo.cs (hunk 10 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/NumberFormatInfo.cs]: Not sure, reverted

File: src/System.Private.CoreLib/shared/System/Globalization/StringInfo.cs (hunk 11 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/StringInfo.cs]: Fair, fixed

File: src/System.Private.CoreLib/shared/System/Globalization/TextInfo.cs (hunk 12 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/TextInfo.cs]: Fixed

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 13]
```diff
@@ -704,12 +700,9 @@ private static CultureData CreateCultureData(string cultureName, bool useUserOve
             culture._sRealName = cultureName;
 
             // Ask native code if that one's real
-            if (culture.InitCultureData() == false)
+            if (!culture.InitCultureData() || !culture.InitCompatibilityCultureData())
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: should be && and not ||

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 14]
```diff
@@ -833,11 +831,11 @@ internal string DisplayName
...
                 if (_sLocalizedDisplayName == null)
                 {
-                    if (this.IsSupplementalCustomCulture)
+                    if (IsSupplementalCustomCulture)
                     {
-                        if (this.IsNeutralCulture)
+                        if (IsNeutralCulture)
                         {
-                            _sLocalizedDisplayName = this.SNATIVELANGUAGE;
+                            _sLocalizedDisplayName = NativeDisplayName;
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: did we rename SNATIVELANGUAGE to NativeDisplayName? should it be NativeLanguageName? I am asking because you renamed SENGLISHLANGUAGE to EnglishLanguageName which is not consistent

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 15]
```diff
@@ -1056,7 +1054,7 @@ internal string ThreeLetterWindowsLanguageName
         /// Localized name for this language (Windows Only) ie: Inglis
         /// This is only valid for Windows 8 and higher neutrals:
         /// </summary>
-        internal string SLOCALIZEDLANGUAGE
+        private string LocalizedDisplayName
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: I prefer to keep the "Language" in the name here

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 16]
```diff
@@ -1100,7 +1098,7 @@ internal string SENGLISHLANGUAGE
         /// <summary>
         /// Native name of this language (Windows Only) ie: Deutsch
         /// </summary>
-        internal string SNATIVELANGUAGE
+        private string NativeDisplayName
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: I prefer to keep the "Language" in the name here

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 17]
```diff
@@ -2325,14 +2321,13 @@ private static int IndexOfTimePart(string format, int startIndex, string timePar
 
         internal static bool IsCustomCultureId(int cultureId)
```
Comment (tarekgh) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: convert this to =>

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.Windows.cs [hunk 18]
```diff
@@ -66,9 +66,9 @@ private unsafe bool InitCultureData()
 
             int result;
             string realNameBuffer = _sRealName;
-            char* pBuffer = stackalloc char[LOCALE_NAME_MAX_LENGTH];
+            char* pBuffer = stackalloc char[LocaleNameMaxLength];
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.Windows.cs]: LOCALE_NAME_MAX_LENGTH is the exact name of a Windows const.  It should really be an one of the interop files, but for now at least it shouldn't have its casing changed.

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.Windows.cs [hunk 19]
```diff
@@ -308,7 +308,7 @@ private string GetLanguageDisplayName(string cultureName)
                 ((ci = GetUserDefaultCulture()) != null) &&
                 !CultureInfo.DefaultThreadCurrentUICulture.Name.Equals(ci.Name))
             {
-                return SNATIVEDISPLAYNAME;
+                return NativeName;
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.Windows.cs]: Same for these; they should maintain the same casing/spelling as the Windows names, as that's what they represent.

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs [hunk 20]
```diff
@@ -19,40 +19,39 @@ namespace System.Globalization
...
+    ///
+    /// sName -- The IETF name of this culture (ie: no sort info, could be neutral)
+    ///                en-US if you pass in en-US
+    ///                en if you pass in en
+    ///                de-DE if you pass in de-DE_phoneb
+    /// </remarks>
     internal partial class CultureData
     {
-        private const int LOCALE_NAME_MAX_LENGTH = 85;
+        private const int LocaleNameMaxLength = 85;
```
Comment (stephentoub) [inline src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: Hmm, I see now it's shared with Unix.  Renaming it might be fine then.

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs (hunk 17 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/CultureData.cs]: Its a bit long IMO for the => format

File: src/System.Private.CoreLib/shared/System/Globalization/CultureData.Windows.cs (hunk 18 above)
Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/CultureData.Windows.cs]: So unix also uses this but I've replaced the occurences of this in the Windows code with interop variables correctly named

**Comment by tarekgh** [comment]
@hughbe looks there is some compilation errors.

```
05:00:27 
shared\System\Globalization\CultureData.Windows.cs(231,122): error CS0103: The name 'LOCALE_NOUSEROVERRIDE' does not exist in the current context [D:\j\workspace\x64_checked_w---43686a05\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
05:00:27 shared\System\Globalization\CultureData.Windows.cs(336,27): error CS0103: The name 'LOCALE_NOUSEROVERRIDE' does not exist in the current context [D:\j\workspace\x64_checked_w---43686a05\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
05:00:27 shared\System\Globalization\CultureData.Windows.cs(516,66): error CS0103: The name 'LOCALE_SISO3166CTRYNAME' does not exist in the current context [D:\j\workspace\x64_checked_w---43686a05\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
05:00:39   SOS.NETCore -> D:\j\workspace\x64_checked_w---43686a05\bin\Product\Windows_NT.x64.Checked\SOS.NETCore.dll
05:00:39 
05:00:39 Build FAILED.
05:00:39 
05:00:39 

```

**Comment by jkotas** [comment]
```
shared\System\Globalization\CultureData.Windows.cs(231,122): error CS0103: The name 'LOCALE_NOUSEROVERRIDE' does not exist in the current context [D:\j\workspace\x64_checked_w---43686a05\src\System.Private.CoreLib\System.Private.CoreLib.csproj]
```

**Comment by tarekgh** [comment]
@hughbe Looks you missed to exclude the test 

```
   System.Globalization.Tests.NumberFormatInfoPercentNegativePattern.PercentNegativePattern_Set_Invalid [FAIL]
05:41:28       Assert.Equal() Failure
05:41:28                 â†“ (pos 0)
05:41:28       Expected: PercentNegativePattern
05:41:28       Actual:   value
05:41:28                 â†‘ (pos 0)
05:41:28       Stack Trace:
05:41:29             at System.AssertExtensions.Throws[T](String paramName, Func`1 testCode)
05:41:29          D:\j\workspace\windows-TGrou---74aa877a\src\System.Globalization\tests\NumberFormatInfo\NumberFormatInfoPercentNegativePattern.cs(54,0): at System.Globalization.Tests.NumberFormatInfoPercentNegativePattern.PercentNegativePattern_Set_Invalid()
```

**Comment by tarekgh** [comment]
Test this please

**Comment by tarekgh** [comment]
@jkotas @stephentoub do you have any other comments here? or we can merge this one. Looks the CI failures here are not related.

File: src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs [hunk 21]
```diff
@@ -2,37 +2,25 @@
...
 {
+    /// <summary>
+    /// This class implements a set of methods for retrieving
+    /// </summary>
     public partial class SortKey
     {
-        //--------------------------------------------------------------------//
-        //                        Internal Information                        //
-        //--------------------------------------------------------------------//
+        internal readonly string _localeName;
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs]: Nit: private ?

File: src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs [hunk 22]
```diff
@@ -2,37 +2,25 @@
...
+    /// </summary>
     public partial class SortKey
     {
-        //--------------------------------------------------------------------//
-        //                        Internal Information                        //
-        //--------------------------------------------------------------------//
+        private readonly string _localeName;
 
-        internal string _localeName;       // locale identifier
+        internal readonly CompareOptions _options;
```
Comment (jkotas) [inline src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs]: Nit: Is it intentionally left as `internal` and not `private`? I am not able to find the place where it is used outside of this file.

Comment (hughbe) [reply src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs]: Yes it is used I found

Comment (jkotas) [reply src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs]: Where is it used? When I change it to private and try to build CoreLib, both Windows and Unix builds work fine.

**Comment by jkotas** [comment]
Could you please resolve the merge conflicts?

File: src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs (hunk 22 above)
Comment (tarekgh) [reply src/System.Private.CoreLib/shared/System/Globalization/SortKey.cs]: I believe this not used outside the class. I tried to search through the source index and couldn't find any hit. I agree with Jan, just change it to private and try to build Windows/Linux.

**Comment by hughbe** [comment]
Merge conflicts fixed and PR feedback incorporated
 Test failure unrelated
> 12:47:18 Also:   hudson.remoting.Channel$CallSiteStackTrace: Remote call to JNLP4-connect connection from 131.107.174.185/131.107.174.185:48920

**Comment by jkotas** [comment]
Thanks!
